### PR TITLE
CVF-7168 upgrade python/nameko support

### PR DESCRIPTION
**Pull request by tomviner**
- Drop support for Python < 3.6.
- Add tests for new Python and Nameko 2.x versions.
- Convert tests to github-actions, as travis free tier is no longer sufficient 

Todo:
- [x] publish to public pypi, once approved

### CODE REVIEW

**Review by gianchub — APPROVED** [review]
LGTM

File: .github/workflows/tox-tests.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install tox~=3.28 'tox-gh-actions<3'
```
[the remark this answers is not in the document]
Comment (tomviner) [reply .github/workflows/tox-tests.yml:25]: Tox v4 is a complete rewrite of the package and requires Python 3.7. 

Also I didn't want to have to check all these https://tox.wiki/en/latest/upgrading.html

The `tox-gh-actions`  restriction, is just for tox 3 compat

**Review by heindsight — APPROVED** [review]

File: tox.ini [hunk 2]
```diff
@@ -1,26 +1,22 @@
 [tox]
-envlist =
-    py27-nameko{2.9, 2.10, 2.11, 2.12, latest},
-    {py35, py36, py37, py38}-nameko{2.7, 2.8, 2.9, 2.10, 2.11, 2.12, latest, next}
-
+envlist = {py36, py37, py38, py39}-nameko{2.12, 2.13, 2.14}
```
Comment (juliotrigo) [inline tox.ini:2]: Does this library work with newer version of Python (3.10, 3.11)? Is that the reason why we're not including them here?

**Review by juliotrigo — APPROVED** [review]

File: tox.ini (hunk 2 above)
Comment (tomviner) [reply tox.ini:2]: Let's try

**Pull request merged**